opty/direct_collocation.py

```diff
@@ -204,7 +204,7 @@ def __init__(self, obj, obj_grad, equations_of_motion, state_symbols,
 
         self.obj_value = []
 
-    def solve(self, free, lagrange=[], zl=[], zu=[]):
+    def solve(self, free, respect_bounds=True, lagrange=[], zl=[], zu=[]):
```

Comment:
```suggestion
    def solve(self, free,lagrange=[], zl=[], zu=[],  respect_bounds=True):
```

Needs to go at end to not break backwards compatibility.

Reply:
> Needs to go at end to not break backwards compatibility.

I did not know this! I thought with kwargs the sequence did not matter.

Reply:
Technically it does because this works: `prob.solve(free, my_lagrange, my_zl, my_zu`. If you don't specify the kwarg name when calling the function/method, it uses the order. Some people may do this.

That's why Python introduced a new syntax, something like `def solve(self, free, *, lagrange=[], ...)` that actually enforces you to name the kwarg on use (I think). But if we add that, it is backwards incompatible.

Reply:
> Technically it does because this works: `prob.solve(free, my_lagrange, my_zl, my_zu`. If you don't specify the kwarg name when calling the function/method, it uses the order. Some people may do this.
> 
> That's why Python introduced a new syntax, something like `def solve(self, free, *, lagrange=[], ...)` that actually enforces you to name the kwarg on use (I think). But if we add that, it is backwards incompatible.

Thanks! My python knowledge still so limited...